src/Build/Graph/ProjectGraph.cs

```diff
@@ -636,18 +636,24 @@ private string DebuggerDisplayString()
                 var targetsToPropagate = ProjectInterpretation.TargetsToPropagate.FromProjectAndEntryTargets(node.ProjectInstance, requestedTargets);
 
                 // Queue the project references for visitation, if the edge hasn't already been traversed.
-                foreach (var referenceNode in node.ProjectReferences)
+                foreach (ProjectGraphNode referenceNode in node.ProjectReferences)
                 {
-                    var applicableTargets = targetsToPropagate.GetApplicableTargetsForReference(referenceNode.ProjectInstance);
+                    ProjectInstance projectInstance = referenceNode.ProjectInstance;
+                    ImmutableList<string> applicableTargets = targetsToPropagate.GetApplicableTargetsForReference(projectInstance);
 
                     if (applicableTargets.IsEmpty)
                     {
                         continue;
                     }
 
-                    var expandedTargets = ExpandDefaultTargets(
+                    if (ProjectInterpretation.IsOuterBuildWithGeneratePackageOnBuildPropertySetToTrue(projectInstance))
```

Comment:
I believe there are a couple issues here.

1. I don't believe "GeneratePackageOnBuild" is "special" here wrt the graph.
2. Consider when A -> B and B multitargets and has `GeneratePackageOnBuild`, then building A without `/graph` would not produce the package. I believe with your changes it would.

There's a little bit of nuance with multitargeting and whatever the entry-points are. ie, outer builds in general only have `GetTargetFrameworks` invoked, unless they're an entry-point in which case their default targets are invoked.